src/Ocelot.Provider.Kubernetes/OcelotBuilderExtensions.cs

```diff
@@ -9,10 +11,23 @@ public static class OcelotBuilderExtensions
     public static IOcelotBuilder AddKubernetes(this IOcelotBuilder builder, bool usePodServiceAccount = true)
     {
         builder.Services
-            .AddKubeClient(usePodServiceAccount)
+            .AddSingleton<IKubeApiClient, KubeApiClient>(KubeApiClientFactory)
             .AddSingleton(KubernetesProviderFactory.Get)
             .AddSingleton<IKubeServiceBuilder, KubeServiceBuilder>()
             .AddSingleton<IKubeServiceCreator, KubeServiceCreator>();
         return builder;
+
+        KubeApiClient KubeApiClientFactory(IServiceProvider sp)
```

Comment:
Factory, factory, factory... 🙈 
If it's a singleton factory, why not move this functionality to a singleton/static `KubernetesProviderFactory` class?
#### FYI
The static `KubernetesProviderFactory` class (aka singleton) will be refactored in the future to become a genuine `IServiceDiscoveryProvider` object through inheritance from the `IServiceDiscoveryProvider` interface.
**Please take note of this future change.**

Reply:
It's just helper method like [original](https://github.com/tintoy/dotnet-kube-client/blob/84b055c885a2afe00781559ea400c0bd8cdfce6d/src/KubeClient.Extensions.DependencyInjection/ClientRegistrationExtensions.cs#L47)used to avoid inlining delegate into AddSingleton. Should I move it to separate class?)

Reply:
Fair enough, not an issue.